beta_rec/datasets/seq_data_utils.py

```diff
@@ -0,0 +1,118 @@
+import numpy as np
+import pandas as pd
+
+import torch
+from torch.utils.data import Dataset
+
+
+def reindex_items(train_data, valid_data=None, test_data=None):
+    train_data = train_data.sort_values(by=['col_user', 'col_timestamp'])
+    test_data = test_data.sort_values(by=['col_user', 'col_timestamp'])
+
+    # train data
+    item_ids = train_data.col_item.unique()
+    n_items = len(item_ids)
+    item2idx = pd.Series(data=np.arange(len(item_ids))+1,index=item_ids)
+    
+    # Build itemmap is a DataFrame that have 2 columns (col_item, item_idx)
+    itemmap = pd.DataFrame({"col_item": item_ids,
+                            'item_idx': item2idx[item_ids].values})
+    train_data = pd.merge(train_data, itemmap, on="col_item", how='inner')
+
+    train_data.col_item = train_data.item_idx
+    train_data = train_data.drop(columns=['item_idx'])
+
+    train_data = train_data.sort_values(by=['col_user', 'col_timestamp'])
+
+    # test data
+    test_data = pd.merge(test_data, itemmap, on="col_item", how='inner')
+    test_data.col_item = test_data.item_idx
+    test_data = test_data.drop(columns=['item_idx'])
+    test_data = test_data.sort_values(by=['col_user', 'col_timestamp'])
+    
+    # valid data
+    if valid_data is not None:
+        valid_data = pd.merge(valid_data, itemmap, on="col_item", how='inner')
+        valid_data.col_item = valid_data.item_idx
+        valid_data = valid_data.drop(columns=['item_idx'])
+        valid_data = valid_data.sort_values(by=['col_user', 'col_timestamp'])
+
+    return train_data, valid_data, test_data
+
+
+def create_seq_db(data):
+    """
+    Convert interactions of a user to a sequence.
+    
+    :param data: the dataset to be transformed
+    """
```

Comment:
Please use the Google Style Python Docstrings. 
https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html